embedded-devices/src/devices/texas_instruments/tmp102/registers.rs

```diff
@@ -0,0 +1,217 @@
+use bondrewd::BitfieldEnum;
+use embedded_devices_derive::device_register;
+use embedded_registers::register;
+use uom::num_rational::Rational32;
+use uom::si::rational32::ThermodynamicTemperature;
+use uom::si::thermodynamic_temperature::degree_celsius;
+
+/// At the end of every conversion, the device updates the temperature
+/// register with the conversion result.
+#[device_register(super::TMP102)]
+#[register(address = 0b0000, mode = "r")]
+#[bondrewd(read_from = "msb0", default_endianness = "be", enforce_bytes = 2)]
+pub struct Temperature {
+    /// The temperature in °C with a resolution of 7.8125m°C/LSB.
+    pub raw_temperature: i16,
+}
+
+/// Temperature alert flag.
+#[derive(BitfieldEnum, Copy, Clone, Default, PartialEq, Eq, Debug, defmt::Format)]
+#[bondrewd_enum(u8)]
+pub enum AlertFlag {
+    /// The alert is unset.
+    #[default]
+    Cleared = 0,
+    /// The associated alert was triggered.
+    Set = 1,
+}
+
+/// Conversion cycle time.
+#[derive(BitfieldEnum, Copy, Clone, Default, PartialEq, Eq, Debug, defmt::Format)]
+#[bondrewd_enum(u8)]
+#[allow(non_camel_case_types)]
+pub enum ConversionCycleTime {
+    /// 4s
+    T_4000 = 0b00,
+    /// 1s
+    T_1000 = 0b01,
+    /// 250ms
+    #[default]
```

Comment:
Small nitpick: None of the enums should derive `Default` since the default value is technically assiocated to the register and not the type. But you can set the default using a recently introduced macro by annotating the register members with `#[register(default = ...)]`.

It would generally be a good idea to annotate the default of each register member separately to ensure we correctly reflect the PoR values.

Example from the INA228 driver:
https://github.com/oddlama/embedded-devices/blob/8d413621addcfd9a75b8e6f193931b9b525847a6/embedded-devices/src/devices/texas_instruments/ina228/registers.rs#L61